posts/ngrx-facade-pattern.md

```diff
@@ -0,0 +1,137 @@
+# Facade Pattern with NgRx
+Over the last few years, I have seen many developers struggle with state management and understanding the [Redux Pattern](https://redux.js.org/understanding/thinking-in-redux/three-principles). By implementing the Facade Pattern, you can add a layer of abstraction over your state management and decouple the knowledge of stores from your component. In this blog, we will discuss the Facade Pattern with NgRx and the benefits of implementing them together.
+
+## What is NgRx?
+NgRx is a reactive state management framework for Angular applications that is based on redux. NgRx provides libraries for managing state, isolation of side effects, entity collection management, Angular router integration, and developer tooling.
+
+### How NgRx Works
+Visit the [NgRx](https://ngrx.io/) website for more resources about the framework. The state management lifecycle below shows how we select and update our read-only state from our store.
+![NgRx Lifecycle](/images/2021/12/ngrx-state-management-lifecycle.png)
+
+### Key Concepts
+* `Actions` describe events that are emitted (dispatched) to change state.
+* State changes are handled by `reducers`, which are pure functions that describe how our state is to be transformed.
+* `Selectors` are pure functions that derive and compose a read-only piece of state.
+* We select a piece of state from the `store` using `selectors`.
+
+## Facade Pattern
+Now we will dive into what is the Facade Pattern and how to implement it with NgRx.
```

Comment:
I would move some words around to get "...dive into what the Facade Pattern is and..."